herbert_core/utilities/general/accumulate_array_to_buffer.m

```diff
@@ -0,0 +1,79 @@
+function [buffer, nel] = accumulate_array_to_buffer (buffer, nel, vals)
+% Append a set of values to a buffer array, doubling its number of elements
+% if there is not enough space left in the buffer in anticipation of further
+% accumulation.
+%
+% Use this rather than the Matlab intrinsic cat function to reduce the amount of
+% memory reallocation should there a large number of accumulations.
+%
+% Append to an exiting array:
+%   >> [buffer, nel] = accumulate_array_to_buffer (buffer, nel, appendvals)
+%
+% To create a buffer:
+%   >> [buffer, nel] = accumulate_array_to_buffer (buffer_size)
+%
+%       Merely a simple utility: creates an array size buffer_size filled with
+%       NaNs and sets nel=0)
+%
+% Input:
+% ------
+%   buffer      Current buffer array
+%   nel         The number of elements currently filled. If [], then assumes
+%               that the buffer is completely full
+%   vals        Array of values to eppend to the buffer
```

Comment:
while the buffer creation has been given lines 12-13, worth putting its argument here as alternatives

Reply:
Done